gateway/reverse_proxy.go

```diff
@@ -535,7 +540,7 @@ func (p *ReverseProxy) ServeHTTPForCache(rw http.ResponseWriter, req *http.Reque
 	return resp
 }
 
-func (p *ReverseProxy) CheckHardTimeoutEnforced(spec *APISpec, req *http.Request) (bool, float64) {
+func CheckHardTimeoutEnforced(spec *APISpec, req *http.Request, logger *logrus.Entry) (bool, float64) {
```

Comment:
Revert this param change and use `log.*` in 553; If there's a *Gateway in APISpec, it may have .Logger() which would be better.

Reply:
Meh, it looks to be an internal one anyway, ideally, can we unexpose it into `checkHardTimeout` ?